web/modules/custom/ingredient_validation/src/Plugin/Validation/Constraint/IngredientConstraintValidator.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\ingredient_validation\Plugin\Validation\Constraint;
+
+use Symfony\Component\Validator\Constraint;
+use Symfony\Component\Validator\ConstraintValidator;
+
+/**
+ * Validates the Ingredients field on the Product CT.
+ */
+class IngredientConstraintValidator extends ConstraintValidator {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function validate($value, Constraint $constraint) {
+    $ingredients = $value->value;
+    $lines = explode('\n', strval($ingredients));
```

Comment:
No need to store `$ingredients` in a variable. You can call `$value->value` in `strval()`.